third_party/move/move-compiler-v2/tools/testdiff/src/main.rs

```diff
@@ -231,8 +224,39 @@ fn classify_tests(
         if !opts.e2e_tests && IGNORED_UNIT_PATHS.iter().any(|p| parent_str.contains(p)) {
             continue;
         }
-        if let (Some(file_name), Some(parent_name)) = (test.file_name(), parent.file_name()) {
-            let key = TestKey::new(parent_name.to_string_lossy(), file_name.to_string_lossy());
+        if let Some(parent_name) = parent.file_name() {
+            let parent_str = parent_name.to_string_lossy().to_string();
+            let (parent_str, file_name_str) =
+                if let Some(captures) = split_match.captures(&parent_str) {
+                    // Treat a parent_str="base.ext"  as a directive
+                    // to map a file "foo.move" to "foo.unit_test.move"
+                    // with new_parent="base".
+                    // See "unit_test.unit_test" for an example.
+                    let base = captures.get(1).unwrap().as_str();
+                    let ext = captures.get(2).unwrap().as_str();
+                    let new_parent = base.to_string();
+                    let mut new_file_name = PathBuf::from(file_name);
+                    new_file_name.set_extension(ext); // replace .move with .ext
+                    let file_with_ext_move = new_file_name.to_string_lossy() + ".move";
+                    if DEBUG {
+                        eprintln!(
+                            "Parent {} matches, new_parent={}, file_with_ext={}",
+                            parent_str, new_parent, file_with_ext_move
+                        );
+                    }
+                    (new_parent, file_with_ext_move.to_string())
+                } else {
+                    (parent_str, file_name.to_string_lossy().to_string())
+                };
+            let key = TestKey::new(parent_str, file_name_str);
+            if DEBUG {
```

Comment:
For consisency, could we do what `warn()` does below, and have something like `debug()`, hiding the `if DEBUG` in various places (minimize some code dup, obscures reading of the code a bit)?

Reply:
Annoyingly, unless we define a macro or have an `if` we can't avoid evaluating the parameters each time.  I don't feel ready to start creating macros in ordinary Rust code.  See the V1 compiler code.

Reply:
Agree about not creating macro for this.